apps/charterafrica/src/components/SearchInput/SearchInput.js

```diff
@@ -0,0 +1,61 @@
+import IconButton from "@mui/material/IconButton";
+import SvgIcon from "@mui/material/SvgIcon";
+import TextField from "@mui/material/TextField";
+import React from "react";
+
+import SearchIcon from "@/charterafrica/assets/icons/Type=search, Size=16, Color=Black.svg";
+import { neutral } from "@/charterafrica/colors";
+
+const InputSearch = React.forwardRef(function InputSearch(props, ref) {
+  const { InputProps, onClick, onMouseDown, ...other } = props;
+  const handleMouseDownSearch = (e) => {
+    e.preventDefault();
+  };
+
+  const endAdornment = (
+    <IconButton
+      aria-label="search"
+      onClick={onClick}
+      onMouseDown={onMouseDown || handleMouseDownSearch}
+      edge="end"
+    >
+      <SvgIcon
+        component={SearchIcon}
+        viewBox="0 0 16 16"
+        sx={{
+          fill: "none",
+          fontSize: "16px",
+        }}
+      />
+    </IconButton>
+  );
+  return (
+    <TextField
+      InputProps={{
+        endAdornment,
+        ...InputProps,
+        placeholder: "Search",
+        size: "small",
+      }}
+      sx={{
+        minWidth: { xs: "auto", sm: "140px" },
+        width: { xs: "auto", sm: "140px" },
+        color: neutral[400],
+        backgroundColor: neutral[800],
+        borderColor: neutral[400],
+        border: "1px solid",
+        borderRadius: "4px",
+        input: {
+          "&::placeholder": {
+            color: neutral[400],
+          },
+        },
+      }}
```

Comment:
Similar to how we customized `Button` by customizing [`theme`](https://github.com/CodeForAfrica/ui/blob/5f3583f4067ef776d30230cf2d1861ea80cfe8c6/apps/charterafrica/src/theme/index.js#L157-L197), can't we do all `Textfield` [customization](https://www.figma.com/file/gbnaO1FYPq16V2gAFbLOkl/CfA---EPD-Charter-Africa---Sep-2022?node-id=5138%3A3775&t=GfFlfQF9yJpnE8Rt-0) in the theme as well?

Reply:
@kilemensi I have given this a try, and got some of it working. What do you think about it? Is it the right approach of do I have to tweak something?

Reply:
Yep, I think we're on the right track @kelvinkipruto . See my comment [here](https://github.com/CodeForAfrica/ui/pull/268#discussion_r1031408726).